### PR TITLE
Origin/doc update 2.0

### PR DESCRIPTION
**Pull request by srinia6**
Update readthedocs from 1.1 to current. @heckj @keedya @BillyAbildgaard @brianparry @dalebremner @RackHD/corecommitters

### CODE REVIEW

**Comment by keedya** [comment]
:+1:

File: docs/rackhd2.0/graphs_current.rst [hunk 1]
```diff
@@ -328,50 +328,50 @@ The following are common API commands that can be used when running the *on-http
...
-        curl <server>/api/1.1/nodes/<id>/workflows/active
+        curl <server>/api/current/nodes/<id>/workflows?active=true
 
 
 **Cancel or Kill an Active Graph running against a Node**
 
 ::
 
-        DELETE /api/1.1/nodes/<id>/workflows/active
+        PUT /api/current/nodes/<id>/workflows/action
```
Comment (BillyAbildgaard) [inline docs/rackhd2.0/graphs_current.rst]: should add that we also need to include
```javascript
{"command": "cancel"} 
```
in the body of this PUT now

File: docs/rackhd2.0/install_os_current.rst [hunk 2]
```diff
@@ -51,14 +51,14 @@ Check the workflow is active or inactive:
...
-    curl <server>/api/1.1/nodes/<identifier>/workflows/active
+    curl <server>/api/current/nodes/<identifier>/workflows/active
 
 
 Stop the active workflow to cancel OS installation:
 
 .. code-block:: REST
 
-    curl -X DELETE <server>/api/1.1/nodes/<identifier>/workflows/active
+    curl -X DELETE <server>/api/current/nodes/<identifier>/workflows/active
```
Comment (BillyAbildgaard) [inline docs/rackhd2.0/install_os_current.rst]: Similar to /workflows, the way you canel an active workflow on a node is by using the PUT method 
```
node/:id/workflows/action
```
and pass
```javascript
{"command": "cancel"} 
in the body of this PUT now
```

File: docs/rackhd2.0/install_os_current.rst [hunk 3]
```diff
@@ -51,14 +51,14 @@ Check the workflow is active or inactive:
 
 .. code-block:: REST
 
-    curl <server>/api/1.1/nodes/<identifier>/workflows/active
+    curl <server>/api/current/nodes/<identifier>/workflows/active
```
Comment (BillyAbildgaard) [inline docs/rackhd2.0/install_os_current.rst]: the way we get active workflow on a node changed to pass in a query

```javascript
/api/current/nodes/<identifier>/workflows?active=true
```

File: docs/rackhd2.0/secure_erase_current.rst [hunk 4]
```diff
@@ -25,14 +25,14 @@ Use below command to check the workflow is active or inactive:
 
 .. code-block:: REST
 
-    curl <server>/api/1.1/nodes/<identifier>/workflows/active
+    curl <server>/api/current/nodes/<identifier>/workflows/active
```
Comment (BillyAbildgaard) [inline docs/rackhd2.0/secure_erase_current.rst]: same comment from above about getting workflow on node using query method applies here

File: docs/rackhd2.0/secure_erase_current.rst [hunk 5]
```diff
@@ -25,14 +25,14 @@ Use below command to check the workflow is active or inactive:
...
 
 
 Use below command to stop the active workflow to cancel secure erase workflow:
 
 .. code-block:: REST
 
-    curl -X DELETE <server>/api/1.1/nodes/<identifier>/workflows/active
+    curl -X DELETE <server>/api/current/nodes/<identifier>/workflows/active
```
Comment (BillyAbildgaard) [inline docs/rackhd2.0/secure_erase_current.rst]: same comment from above about using PUT to delete with nodes/id/workflows/action applies here

**Comment by benbp** [comment]
Just a general question in response to @BillyAbildgaard's comments, is this a blanket find and replace or is it validated that all these documented routes work as well, since we've made a lot of backwards incompatible API changes from 1.1 to 2.0?

**Comment by keedya** [comment]
@benbp That is suppose to be initiating the new documentation for 2.0 and validate the existing routes. There will be more PRs going forward to list all the APIs in 2.0 that aren't mentioned in the current guides

File: docs/rackhd2.0/graphs_current.rst (hunk 1 above)
Comment (yyscamper) [inline docs/rackhd2.0/graphs_current.rst]: Agree

File: docs/rackhd2.0/nodes_current.rst [hunk 6]
```diff
@@ -119,10 +119,10 @@ Sample Output:
 
 .. code-block:: REST
 
-    curl -X PATCH \
+    curl -X PUT \
```
Comment (yyscamper) [inline docs/rackhd2.0/nodes_current.rst]: I think this doesn't work now since we have migrated to new obm model.

**Comment by heckj** [comment]
Key question: Do we want to encourage the use of `/api/current` vs. `/api/2.0`

I'd recommend not doing that, and being specific about API version, continuing to leverage semantic versioning specifically. We'll be shortly encouraging people to rewrite code from the 1.1 API to the 2.0 API set for clients to RackHD, and not all of them are using the swagger generated pieces and going from there. When they don't use code gen tools, they look at examples, and all these examples using `/api/current` as a shorthand, but the specific meaning at /api/current is going to change over extended time.

I think it's useful as a shorthand, and was an interesting alias early on, but not something that I'd recommend we continue - maybe continuing to use `/api/current` for southbound interfaces if anything, but realistically we should be encouraging specific API contracts and API versioning to go with it, so my preference would be that we just don't even advertise we have `/api/current` and work towards deprecating it, and change all these API references to `/api/2.0/` to be explicit about it

**Comment by yyscamper** [comment]
I totally agree with @heckj , use /api/2.0 and deprecate the /api/current & /api/common.  The /api/current & /api/common is going to be changed over time. Image we will define API 3.0 in future, it will be a huge work to do API format replacement.

**Comment by anhou** [comment]
Also agree using /api/2.0 /api/1.1/ explicitly, /current is not directly and need to change when version change.  for example, if using 'current', when 3.0 version come, not only add 3.0 docs, 2.0 docs's 'current' also need to change to '2.0' again.

**Comment by keedya** [comment]
@heckj we had the impression that we only supporting one guide once we deprecate1.1 version. That is said to avoid a big change like this in the future, we decided to use current instead. Are we planning on full guide for every version?

**Comment by keedya** [comment]
In my opinion, once we make this change, going forward, we should only document the changes in different versions

**Comment by BillyAbildgaard** [comment]
@heckj @benbp @yyscamper @anhou 

With Andre's thought process, does it make sense to use /2.0 for long term?  I think we should make it explicitly clear that 2.0 is the new "current" once we turn it on as the default.  Are we planning to change the documentation every time there is a semantic versioning bump?

/2.0.1, /2.1, etc?

**Comment by BillyAbildgaard** [comment]
Just added a commit that deletes the RackHD 2.0 user's guide and adds 2.0 API commands to the existing User's Guide when there's differences

**Comment by keedya** [comment]
:+1

File: docs/rackhd/install_os.rst [hunk 7]
```diff
@@ -53,13 +53,26 @@ Check the workflow is active or inactive:
 
     curl <server>/api/1.1/nodes/<identifier>/workflows/active
 
+2.0 API Check the workflow is active or inactive:
```
Comment (heckj) [inline docs/rackhd/install_os.rst]: This is requesting what active workflows are against the node if I'm reading this correctly?

File: docs/rackhd/install_os.rst [hunk 8]
```diff
@@ -53,13 +53,26 @@ Check the workflow is active or inactive:
...
+
+    curl <server>/api/2.0/nodes/<identifier>/workflows?active=true
 
 Stop the active workflow to cancel OS installation:
 
 .. code-block:: REST
 
     curl -X DELETE <server>/api/1.1/nodes/<identifier>/workflows/active
 
+2.0 API Stop the active workflow to cancel OS installation:
```
Comment (heckj) [inline docs/rackhd/install_os.rst]: maybe better as 

    2.0 API Stop the currently active workflow for a node

**Comment by heckj** [comment]
included a few wording update suggestions, but regardless 👍

**Comment by BillyAbildgaard** [comment]
@heckj made your recommended changes.  I agree they have better wording than what I used

**Comment by heckj** [comment]
👍

**Comment by anhou** [comment]
:+1:

**Pull request merged**